File: lc_soln/0049_group_anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string, std::vector<std::string>> variants;
        for (const std::string& s : strs) {
            std::vector<int> freq(26, 0);
            for (char c : s) {
                freq[c - 'a']++;
            }
            std::string key = "";
            for (int f : freq) {
                key += '#' + std::to_string(f);
            }
            variants[key].push_back(s);
        }
        std::vector<std::vector<std::string>> result;
        result.reserve(variants.size());
        for (const auto& [_, val] : variants) {
            result.push_back(std::move(val));
        }
        return result;
    }
};

int main() {
    return 0;
}

File: lc_soln/0049_group_anagrams/solution.cs
public class Solution {
    public List<List<string>> GroupAnagrams(string[] strs) {
        var variants = new Dictionary<string, List<string>>();
        foreach (string s in strs) {
            var freq = new int[26];
            foreach (char c in s) {
                freq[c - 'a']++;
            }
            string key = string.Join("#", freq);
            if (!variants.ContainsKey(key)) {
                variants[key] = new List<string>();
            }
            variants[key].Add(s);
        }
        return new List<List<string>>(variants.Values);
    }
}
